pkg/tsdb/azuremonitor/macros_test.go

```diff
@@ -0,0 +1,155 @@
+			expected:  "1 == 1",
+			Err:       require.NoError,
+		},
+		{
+			name:      "$__timeFilter has no column parameter should use default time field",
+			timeRange: timeRange,
+			query:     &tsdb.Query{},
+			kql:       "$__timeFilter()",
+			expected:  "['TimeGenerated'] >= datetime('2018-03-15T13:00:00Z') and ['TimeGenerated'] <= datetime('2018-03-15T13:34:00Z')",
+			Err:       require.NoError,
+		},
+		{
+			name:      "$__timeFilter has time field parameter",
+			timeRange: timeRange,
+			query:     &tsdb.Query{},
+			kql:       "$__timeFilter(myTimeField)",
+			expected:  "['myTimeField'] >= datetime('2018-03-15T13:00:00Z') and ['myTimeField'] <= datetime('2018-03-15T13:34:00Z')",
+			Err:       require.NoError,
+		},
+		{
+			name:      "$__timeFrom and $__timeTo is in the query and range is a specific interval",
+			timeRange: timeRange,
+			query:     &tsdb.Query{},
+			kql:       "myTimeField >= $__timeFrom() and myTimeField <= $__timeTo()",
+			expected:  "myTimeField >= datetime('2018-03-15T13:00:00Z') and myTimeField <= datetime('2018-03-15T13:34:00Z')",
+			Err:       require.NoError,
+		},
+		{
+			name:      "$__interval should use the defined interval from the query",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				Model: simplejson.NewFromAny(map[string]interface{}{
+					"interval": "5m",
+				}),
+			},
+			kql:      "bin(TimeGenerated, $__interval)",
+			expected: "bin(TimeGenerated, 300000ms)",
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__interval should use the default interval if none is specified",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      "bin(TimeGenerated, $__interval)",
+			expected: "bin(TimeGenerated, 34000ms)",
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable should replace values with KQL style escaped strings",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      `CounterPath in ($__escapeMulti('\\grafana-vm\Network(eth0)\Total','\\grafana-vm\Network(eth1)\Total'))`,
+			expected: `CounterPath in (@'\\grafana-vm\Network(eth0)\Total', @'\\grafana-vm\Network(eth1)\Total')`,
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable and has one selected value that contains comma",
+			timeRange: timeRange,
+			query: &tsdb.Query{
+				DataSource: &models.DataSource{},
+				Model:      simplejson.NewFromAny(map[string]interface{}{}),
+			},
+			kql:      `$__escapeMulti('\\grafana-vm,\Network(eth0)\Total Bytes Received')`,
+			expected: `@'\\grafana-vm,\Network(eth0)\Total Bytes Received'`,
+			Err:      require.NoError,
+		},
+		{
+			name:      "$__escapeMulti with multi template variable and is not wrapped in single quotes should fail",
```

Comment:
should this return an error instead of silently failing

Reply:
What is failing silently? `$__escapeMulti with multi template variable and is not wrapped in single quotes should fail` tests that it returns an error.

Reply:
Argh my bad. I misread the `Err:      require.Error,` line.
I'm not used to checking for errors like that.